arangod/Pregel/Worker/Worker.cpp

```diff
@@ -352,18 +393,19 @@ void Worker<V, E, M>::_initializeVertexContext(VertexContext<V, E, M>* ctx) {
   ctx->_gss = _config->globalSuperstep();
   ctx->_lss = _config->localSuperstep();
   ctx->_context = _workerContext.get();
-  ctx->_quiver = _quiver;
+  ctx->_quivers = _quivers;
   ctx->_readAggregators = _workerContext->_readAggregators.get();
 }
 
 // internally called in a WORKER THREAD!!
 template<typename V, typename E, typename M>
-bool Worker<V, E, M>::_processVertices() {
+futures::Future<Result> Worker<V, E, M>::_processVertices(
+    size_t idx, std::shared_ptr<Quiver<V, E>> quiver) {
   double start = TRI_microtime();
 
   // thread local caches
-  InCache<M>* inCache = _inCaches[0];
-  OutCache<M>* outCache = _outCaches[0];
+  InCache<M>* inCache = _inCaches[idx];
+  OutCache<M>* outCache = _outCaches[idx];
```

Comment:
Can you make the two caches a parameter to `_processVertices` too and get rid of that `idx` parameter?